src/librustdoc/html/render/mod.rs

```diff
@@ -2506,7 +2507,8 @@ fn render_call_locations<W: fmt::Write>(mut w: W, cx: &mut Context<'_>, item: &c
         // Look for the example file in the source map if it exists, otherwise return a dummy span
         let file_span = (|| {
             let source_map = tcx.sess.source_map();
-            let crate_src = tcx.sess.local_crate_source_file()?;
+            let crate_src =
+                tcx.sess.local_crate_source_file(RemapPathScopeComponents::DIAGNOSTICS)?;
```

Comment:
This is being compared against `RealFileName::LocalPath` below, that is without any remapping. So we should not remap here either. Can you make `local_crate_source_file` return a `RealFileName`? And since we have the session available there, we probably don't need to pass in the scopes.

Reply:
The logic here explicitly takes the remapping into account, always returning the "real-path" would change this, for better or worth. @GuillaumeGomez would rustdoc prefer to always have "real path" or does rustdoc wants to take into account the remapping here?

As for returning `RealFileName` we could probably do that, not sure if it's a good idea since all the caller seems to want to `PathBuf`/`Path`.

Reply:
> The logic here explicitly takes the remapping into account, always returning the "real-path" would change this

Yes, it looks like the other side of the comparison is also coming from the source map and never expects to that no remapping has happened. 

> As for returning RealFileName we could probably do that, not sure if it's a good idea since all the caller seems to want to PathBuf/Path.

Returning `RealFileName` would retain complete information and push the decision of which PathBuf to use to the usage site. I think that's a good way to do it and would be consistent with other places, like this for example:
https://github.com/rust-lang/rust/blob/8579a1856a868033b792c08e5c73a7c9dbe4d273/compiler/rustc_session/src/options.rs#L224